content/en/blog/_posts/2023-05-05-memory-qos-cgroups-v2/index.md

```diff
@@ -0,0 +1,286 @@
+
+  In cgroups v2, `memory.max` is analogous to `memory.limit_in_bytes` in cgroupv1.
+  Memory QoS maps `memory.max` to `spec.containers[].resources.limits.memory` to
+  specify the hard limit for memory usage. If the memory consumption goes above this
+  level, the kernel invokes its OOM Killer.
+
+  cgroups v2 also added `memory.high` configuration . Memory QoS uses `memory.high`
+  to set memory usage throttle limit. If the `memory.high` limit is breached,
+  the offending cgroups are throttled, and the kernel tries to reclaim memory
+  which may avoid an OOM kill.
+
+## How it works
+
+### Cgroups v2 memory controller interfaces & Kubernetes container resources mapping
+
+Memory QoS uses the memory controller of cgroups v2 to guarantee memory resources in
+Kubernetes. cgroupv2 interfaces that this feature uses are: 
+* `memory.max`
+* `memory.min`
+* `memory.high`.
+
+{{< figure src="/blog/2023/05/05/qos-memory-resources/memory-qos-cal.svg" title="Memory QoS Levels" alt="Memory QoS Levels" >}}
+
+`memory.max` is mapped to memory limit specified in the Pod spec. The kubelet and 
+the container runtime enforce the limit and prevent the container from using more
+than the configured resource limit. If a process in a container tries to consume
+more than the specified limit, kernel terminates the process with an out of
+memory Out of Memory (OOM) error. 
+
+{{< figure src="/blog/2023/05/05/qos-memory-resources/container-memory-max.svg" title="`memory.max` maps to `limits.memory`" alt="`memory.max` maps to `limits.memory`" >}}
+
+`memory.min` is mapped to memory `requests`, which results in reservation of memory resources
+that can never reclaimed by the kernel. This is how Memory QoS ensures the availability of 
+memory for Kubernetes pods. If there's no unprotected reclaimable memory available, OOM 
+killer is invoked.
+
+{{< figure src="/blog/2023/05/05/qos-memory-resources/container-memory-min.svg" title="`memory.min` maps to `requests.memory`" alt="`memory.min` maps to `requests.memory`" >}}
+
+For memory protection, in addition to the original way of limiting memory usage, Memory QoS
+throttles workload approaching its memory limit, ensuring that the system is not overwhelmed
+by sporadic increases in memory usage. A new field, `memoryThrottlingFactor`, is available in
+the KubeletConfiguration when you enable MemoryQoS feature. It is set to 0.9 by default. 
+`memory.high` is mapped to throttling limit calculated by using `memoryThrottlingFactor`,
+`spec.containers[].resources.requests.memory` and `spec.containers[].resources.limits.memory`
+as in the formula below, and rounding down the value to the nearest page size:
+
+{{< figure src="/blog/2023/05/05/qos-memory-resources/container-memory-high.svg" title="`memory.high` formula" alt="`memory.high` formula" >}} 
+
+**Note**: If a container has no memory limits specified, we substitute `limits.memory` for node allocatable memory.
```

Comment:
Can you rephrase to [avoid using “we”](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-using-we)?